src/rest_vol.c

```diff
@@ -2732,14 +2477,258 @@ RV_copy_object_loc_info_callback(char *HTTP_response, void *callback_data_in, vo
         yajl_tree_free(parse_tree);
 
     if (ret_value < 0) {
-        free(GCPL_buf);
+        RV_free(GCPL_buf);
         GCPL_buf                  = NULL;
         loc_info_out->GCPL_base64 = NULL;
     }
 
     return ret_value;
 } /* end RV_copy_object_loc_info_callback() */
 
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_link_name_by_index
+ *
+ * Purpose:     This callback is used to copy the name of an link
+ *              in the server's response by index. It allocates heap memory for the name,
+ *              and returns a pointer to the memory by callback_data_out.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_copy_link_name_by_index(char *HTTP_response, void *callback_data_in, void *callback_data_out)
+{
+    yajl_val           parse_tree = NULL, key_obj = NULL, link_obj = NULL;
+    const char        *parsed_link_name   = NULL;
+    char              *parsed_link_buffer = NULL;
+    H5VL_loc_by_idx_t *idx_params         = (H5VL_loc_by_idx_t *)callback_data_in;
+    hsize_t            index              = idx_params->n;
+    char             **link_name          = (char **)callback_data_out;
+    const char        *curr_key           = NULL;
+    herr_t             ret_value          = SUCCEED;
+
```

Comment:
`idx_params` should also be checked as non-NULL here and `index` shouldn't be assigned until it's been verified that `idx_params` is non-NULL.